publication/src/main/scala/hmda/publication/reports/disclosure/D51.scala

```diff
@@ -0,0 +1,155 @@
+package hmda.publication.reports.disclosure
+
+import java.util.Calendar
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import hmda.model.publication.reports.ApplicantIncomeEnum._
+import hmda.publication.reports._
+import hmda.model.publication.reports.ReportTypeEnum.Disclosure
+import hmda.model.publication.reports._
+import hmda.publication.reports.util.DateUtil._
+import hmda.publication.reports.util.DispositionType._
+import hmda.publication.reports.util.ReportUtil._
+import hmda.query.model.filing.LoanApplicationRegisterQuery
+
+import scala.concurrent.Future
+
+case class D51(
+  respondentId: String,
+  institutionName: String,
+  table: String,
+  reportType: ReportTypeEnum,
+  description: String,
+  year: Int,
+  reportDate: String,
+  msa: MSAReport,
+  applicantIncomes: List[ApplicantIncome],
+  total: List[Disposition]
```

Comment:
I agree that's a good idea. I'm not totally familiar with all of the A&D reports, so I don't have a sense for which ones/how many will be repeated. How would you feel about waiting until we've done a few more, then using that knowledge to make the best optimization?